File: Week 1/task3.py
""" Необычные шахматы от BPS """
while True:
    COL_1 = int(input())
    STR_1 = int(input())
    COL_2 = int(input())
    STR_2 = int(input())
    if COL_1 > 0 and STR_1 > 0 and COL_2 > 0 and STR_2 > 0:
        break
if ((COL_1 + 2 == COL_2 and STR_1 + 5 == STR_2) or (COL_1 + 5 == COL_2 and STR_1 + 2 == STR_2)):
    print("YESSSS!")
elif ((COL_2 + 2 == COL_1 and STR_2 + 5 == STR_1) or (COL_2 + 5 == COL_1 and STR_2 + 2 == STR_1)):
    print("YESSSS!")
else:
    print("No no")

File: HW4/task3.go
//Необычные шахматы от BPS
package main

import "fmt"

func main() {
	var col_1, str_1, col_2, str_2 int
	fmt.Scan(&col_1)
	fmt.Scan(&str_1)
	fmt.Scan(&col_2)
	fmt.Scan(&str_2)
	if (col_1+2 == col_2 && str_1+5 == str_2) || (col_1+5 == col_2 && str_1+2 == str_2) {
		fmt.Print("YESSSS!")
	} else if (col_2+2 == col_1 && str_2+5 == str_1) || (col_2+5 == col_1 && str_2+2 == str_1) {
		fmt.Print("YESSSS!")
	} else {
		fmt.Print("No no")
	}
}
